packages/data-context/test/unit/sources/ProjectDataSource.spec.ts

```diff
@@ -189,3 +189,92 @@ describe('findSpecs', () => {
     expect(specs).to.have.length(3)
   })
 })
+
+describe('getDefaultSpecFileName', () => {
+  context('dirname', () => {
+    it('returns pattern without change if it is do not a glob', () => {
+      const specPattern = 'cypress/e2e/foo.spec.ts'
+      const defaultFileName = getDefaultSpecFileName(specPattern)
+
+      expect(defaultFileName).to.eq(specPattern)
+    })
+
+    it('remove ** from glob if it is not in the beginning', () => {
+      const defaultFileName = getDefaultSpecFileName('cypress/**/foo.spec.ts')
+
+      expect(defaultFileName).to.eq('cypress/foo.spec.ts')
+    })
+
+    it('replace ** for cypress if it starts with **', () => {
+      const defaultFileName = getDefaultSpecFileName('**/e2e/foo.spec.ts')
+
+      expect(defaultFileName).to.eq('cypress/e2e/foo.spec.ts')
+    })
+
+    it('replace ** for cypress if it starts with ** and omit extra **', () => {
+      const defaultFileName = getDefaultSpecFileName('**/**/foo.spec.ts')
+
+      expect(defaultFileName).to.eq('cypress/foo.spec.ts')
+    })
+
+    it('selects first option if there are multiples possibilities of values', () => {
+      const defaultFileName = getDefaultSpecFileName('{cypress,tests}/{integration,e2e}/foo.spec.ts')
+
+      expect(defaultFileName).to.eq('cypress/integration/foo.spec.ts')
+    })
+  })
+
+  context('filename', () => {
+    it('replace * for filename', () => {
+      const defaultFileName = getDefaultSpecFileName('cypress/e2e/*.spec.ts')
+
+      expect(defaultFileName).to.eq('cypress/e2e/filename.spec.ts')
+    })
+
+    it('selects first option if there are multiples possibilities of values', () => {
+      const defaultFileName = getDefaultSpecFileName('cypress/e2e/{foo,filename}.spec.ts')
+
+      expect(defaultFileName).to.eq('cypress/e2e/foo.spec.ts')
+    })
+  })
+
+  context('test extension', () => {
+    it('replace * for filename', () => {
+      const defaultFileName = getDefaultSpecFileName('cypress/e2e/filename.*.ts')
+
+      expect(defaultFileName).to.eq('cypress/e2e/filename.cy.ts')
+    })
+
+    it('selects first option if there are multiples possibilities of values', () => {
+      const defaultFileName = getDefaultSpecFileName('cypress/e2e/filename.{spec,cy}.ts')
+
+      expect(defaultFileName).to.eq('cypress/e2e/filename.spec.ts')
+    })
+  })
+
+  context('lang extension', () => {
+    it('if project use TS, set TS as extension if it exists in the glob', () => {
+      const defaultFileName = getDefaultSpecFileName('cypress/e2e/filename.cy.ts', 'ts')
+
+      expect(defaultFileName).to.eq('cypress/e2e/filename.cy.ts')
+    })
+
+    it('if project use TS, set TS as extension if it exists in the options of extensions', () => {
+      const defaultFileName = getDefaultSpecFileName('cypress/e2e/filename.cy.{js,ts,tsx}', 'ts')
+
+      expect(defaultFileName).to.eq('cypress/e2e/filename.cy.ts')
+    })
+
```

Comment:
Nice job on these tests, super useful and makes the actual implementation much clearer.